xarray/core/dataset.py

```diff
@@ -3529,7 +3531,8 @@ def shift(self, shifts=None, **shifts_kwargs):
             if name in self.data_vars:
                 var_shifts = dict((k, v) for k, v in shifts.items()
                                   if k in var.dims)
-                variables[name] = var.shift(**var_shifts)
+                variables[name] = var.shift(
+                    fill_value=fill_value, **var_shifts)
```

Comment:
side note: we should probably pass `var_shifts` by position rather than as keyword arguments, so we can support non-string keys.